harness/determined/common/api/_util.py

```diff
@@ -135,3 +135,130 @@ def _task_is_done_loading() -> Tuple[bool, Optional[str]]:
+    session: api.Session,
+    role_name: str,
+    workspace_name: Optional[str] = None,
+    username_to_assign: Optional[str] = None,
+    group_name_to_assign: Optional[str] = None,
+) -> None:
+    user_assign, group_assign = create_assignment_request(
+        session=session,
+        role_name=role_name,
+        workspace_name=workspace_name,
+        username_to_assign=username_to_assign,
+        group_name_to_assign=group_name_to_assign,
+    )
+    req = bindings.v1AssignRolesRequest(
+        userRoleAssignments=user_assign, groupRoleAssignments=group_assign
+    )
+    bindings.post_AssignRoles(session, body=req)
+
+    scope = " globally"
+    if workspace_name:
+        scope = f" to workspace {workspace_name}"
+    if len(user_assign) > 0:
+        role_id = user_assign[0].roleAssignment.role.roleId
+        print(
+            f"assigned role '{role_name}' with ID {role_id} "
+            + f"to user '{username_to_assign}' with ID {user_assign[0].userId}{scope}"
+        )
+    else:
+        role_id = group_assign[0].roleAssignment.role.roleId
+        print(
+            f"assigned role '{role_name}' with ID {role_id} "
+            + f"to group '{group_name_to_assign}' with ID {group_assign[0].groupId}{scope}"
+        )
+
+
+def usernames_to_user_ids(session: api.Session, usernames: List[str]) -> List[int]:
+    usernames_to_ids: Dict[str, Optional[int]] = {u: None for u in usernames}
+    users = bindings.get_GetUsers(session).users or []
+    for user in users:
+        if user.username in usernames_to_ids:
+            usernames_to_ids[user.username] = user.id
+
+    missing_users = []
+    user_ids = []
+    for username, user_id in usernames_to_ids.items():
+        if user_id is None:
+            missing_users.append(username)
+        else:
+            user_ids.append(user_id)
+
+    if missing_users:
+        raise api.errors.BadRequestException(
+            f"could not find users for usernames {', '.join(missing_users)}"
+        )
+    return user_ids
+
+
+def group_name_to_group_id(session: api.Session, group_name: str) -> int:
+    body = bindings.v1GetGroupsRequest(name=group_name, limit=1, offset=0)
+    resp = bindings.post_GetGroups(session, body=body)
+    groups = resp.groups
+    if groups is None or len(groups) != 1 or groups[0].group.groupId is None:
+        raise api.errors.BadRequestException(f"could not find user group name {group_name}")
+    return groups[0].group.groupId
+
+
+def workspace_by_name(session: api.Session, name: str) -> bindings.v1Workspace:
+    assert name, "workspace name cannot be empty"
+    w = bindings.get_GetWorkspaces(session, nameCaseSensitive=name).workspaces
+    assert len(w) <= 1, "workspace name is assumed to be unique."
+    if len(w) == 0:
+        raise not_found_errs("workspace", name, session)
+    return bindings.get_GetWorkspace(session, id=w[0].id).workspace
+
+
+# not_found_errs mirrors NotFoundErrs from the golang api/errors.go. In the cases where
```

Comment:
I don't get what's going on with this function. Could it be part of your cleanup?

Reply:
Yes, I'll look into it.  I didn't ask that question earlier, I just shuffled it around.

Reply:
looked into it.  It just helps users remember that 404 errors in rbac cases might be permissions errors.

I promoted the comment to a docstring and added more explanation.

Reply:
The docstring is a better place to put the help text, for sure. And thanks for figuring out what the function is for.

I still have to do more work to understand what the function does than I want a person to have to do. Something seems off about the design. Bleh.

But anyway, I think bigger changes are outside the scope of this PR. So maybe here just a suggestion like:
```
"""Construct a NotFoundException from passed strings.

This function creates NotFoundExceptions with the same syntax as the NotFoundErrs from api/errors.go so that messages from both layers look the same. If RBAC is enabled, the exceptions underlying the 404s that this function constructs might be the result of a permissions error. In that event, this function appends a suggestion to the constructed exception string that the user to check permissions.
```

The type mismatch between the actual return type and its superclass is a little weird. That'd be an easy fix if you want to make it.